File: backend/src/main/kotlin/io/github/tyostokarry/bookshelf/controller/advice/ApiResponse.kt
package io.github.tyostokarry.bookshelf.controller.advice

data class ApiResponse<T>(
    val data: T? = null,
    val error: ErrorResponse? = null,
)

data class ErrorResponse(
    val message: String,
    val code: ErrorCodes,
    val fieldErrors: Map<String, String>? = null,
)

File: frontend/src/types/api-response.ts
import type { ApiErrorCode } from "./api-error-codes";

export interface ApiResponse<T> {
  data: T | null;
  error: ApiErrorResponse | null;
}

export interface ApiErrorResponse {
  message: string;
  code: ApiErrorCode;
  fieldErrors?: Record<string, string>;
}
